kbase-extension/static/kbase/js/widgets/function_output/kbaseGenericMatrix.js

```diff
@@ -0,0 +1,308 @@
+/**
+ * Output widget to vizualize ExpressionMatrix object.
+ * Pavel Novichkov <psnovichkov@lbl.gov>
+ * @public
+ */
+
+define ([
+    'uuid',
+    'jquery',
+    'kbwidget',
+    'kbaseAuthenticatedWidget',
+    'kbaseTabs',
+    'narrativeConfig',
+		'kbase-generic-client-api',
+    'bluebird',
+    // For effect
+    'bootstrap',
+    'jquery-dataTables',
+    'datatables.net-buttons',
+    'datatables.net-buttons-bs',
+    'datatables.net-buttons-html5',
+    'datatables.net-buttons-print'
+], function(
+    Uuid,
+    $,
+    KBWidget,
+    kbaseAuthenticatedWidget,
+    kbaseTabs,
+    Config,
+    GenericClient,
+    Promise
+) {
+    'use strict';
+
+    return KBWidget({
+        name: 'kbaseGenericMatrix',
+        parent : kbaseAuthenticatedWidget,
+        version: '1.0.2',
+        options: {
+            matrixID: null,
+            workspaceID: null,
+            loadingImage: Config.get('loading_gif')
+        },
+
+        // Prefix for all div ids
+        pref: null,
+
+        // KBaseFeatureValue client
+        featureValueClient: null,
+
+        // Matrix data to be visualized
+        matrixStat: null,
+
+        init: function(options) {
+            this._super(options);
+            this.pref = this.uuid();
+            // Create a message pane
+            this.$messagePane = $('<div/>').addClass('kbwidget-message-pane kbwidget-hide-message');
+            this.$elem.append(this.$messagePane);
+
+            return this;
+        },
+
+        loggedInCallback: function(event, auth) {
+
+            // error if not properly initialized
+            if (this.options.matrixID == null) {
```

Comment:
Below, it's `this.options.upas.matrixID` that's being used for the search. That should be tested for existence here, too.